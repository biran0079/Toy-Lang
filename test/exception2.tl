try{
  try{
    try {
      throw 123;  
    } catch (e) {
      print("inner catch:",e);
      throw e;
    } finally {
      print("inner finally");
    }
  } catch(e2) {
    print("outer catch:", e2);
  } finally {
    print("outer finally");
  }
}catch(e){
  print("should not reach here");
}finally{
  print("last finally");  
}
