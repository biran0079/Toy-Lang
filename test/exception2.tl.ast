[stmts, [try, [stmts, [try, [stmts, [try, [stmts, [throw, [int, 123]]], [id, e], [stmts, [call, [id, print], [exps, [string, inner catch:], [id, e]]], [throw, [id, e]]], [stmts, [call, [id, print], [exps, [string, inner finally]]]]]], [id, e2], [stmts, [call, [id, print], [exps, [string, outer catch:], [id, e2]]]], [stmts, [call, [id, print], [exps, [string, outer finally]]]]]], [id, e], [stmts, [call, [id, print], [exps, [string, should not reach here]]]], [stmts, [call, [id, print], [exps, [string, last finally]]]]]]
