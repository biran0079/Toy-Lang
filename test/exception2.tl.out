inner catch: 123
inner finally
outer catch: 123
outer finally
last finally
